File: articles/angela-duckworth-and-the-modern-science-of-grit/go/duckworth_grit_score.go
package main

import "fmt"

func GritScore(perseveranceEffort float64, consistencyInterest float64) float64 {
	return 0.60*perseveranceEffort + 0.40*consistencyInterest
}

func main() {
	fmt.Printf("Duckworth-style synthetic grit score: %.3f\n", GritScore(0.82, 0.61))
}

File: articles/angela-duckworth-and-the-modern-science-of-grit/rust/duckworth_grit_score.rs
fn grit_score(perseverance_effort: f64, consistency_interest: f64) -> f64 {
    0.60 * perseverance_effort + 0.40 * consistency_interest
}

fn main() {
    println!(
        "Duckworth-style synthetic grit score: {:.3}",
        grit_score(0.82, 0.61)
    );
}
